### PR TITLE
trigonometry

### PR DESCRIPTION
**Pull request by amosr**
I'm tempted to add modulus, but I'm not sure whether int and double modulus are the same polymorphic operation, or if it should be `mod` and `fmod`

! @jystic @tranma

### CODE REVIEW

File: data/sea/20-simple.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ static idouble_t INLINE iint_extend   (iint_t    x)              { return x; }
 static iint_t    INLINE iint_neg      (iint_t    x)              { return -x; }
 static iint_t    INLINE iint_add      (iint_t    x, iint_t    y) { return x + y; }
 static iint_t    INLINE iint_sub      (iint_t    x, iint_t    y) { return x - y; }
+static iint_t    INLINE iint_mod      (iint_t    x, iint_t    y) { return x % y; }
```
Comment (jystic) [inline data/sea/20-simple.h]: looks like you did add modulus?

Comment (amosr) [reply data/sea/20-simple.h]: only to the backend yet. heh.

**Comment by amosr** [comment]
I'll just add some sanity tests in a sec

File: data/sea/20-simple.h (hunk 1 above)
Comment (jystic) [reply data/sea/20-simple.h]: what's the difference in semantics between the integer and double variants?

**Comment by amosr** [comment]
interesting tidbit: ages ago, when classifying double operations into whether they produced nans or not, atan, atan2, cos, sin, tan and tanh are all "clean". it feels more consistent to just say all double operations are possibly, though. any preference?

**Comment by amosr** [comment]
I'm not confident enough in my double testing code to say for sure that all tans are valid. I believe that `tan (pi/2) = sin (pi/2) / cos (pi/2) = 1 / 0`, so it *should* be able to produce infinity, even though for me `cos (pi/2) = 6e-17`.

I don't think we lose anything by making them Possibly, so I'll keep it.

**Comment by amosr** [comment]
I realised integer modulo is a somewhat larger job, because it would need to be in Possibly (1 `mod` 0). the machinery for this is already in place for doubles, but would be a bit different for integers.

so, given we can already encode modulo clumsily, I will leave both modulo operations for later

**Comment by amosr** [comment]
going to hold off on merging this, because if anybody is using any of these as variable names it will break. I had a look but couldn't see any.

**Pull request merged**